src/wallet/api.rs

```diff
@@ -1267,193 +1270,237 @@ impl Wallet {
             })
-            .collect::<Vec<_>>();
+            .collect();
+        let available_swap_utxos: Vec<_> = swap_utxos
+            .into_iter()
+            .filter(|(utxo, _)| {
+                let outpoint = OutPoint::new(utxo.txid, utxo.vout);
+                !locked_utxos.contains(&outpoint)
+            })
+            .collect();
 
-        if unspents.is_empty() {
-            log::warn!("No spendable UTXOs available, returning empty selection with 0 sats");
-            return Ok(Vec::new());
-        }
+        // Calculate totals for each type
+        let regular_total: u64 = available_regular_utxos
+            .iter()
+            .map(|(utxo, _)| utxo.amount.to_sat())
+            .sum();
+        let swap_total: u64 = available_swap_utxos
+            .iter()
+            .map(|(utxo, _)| utxo.amount.to_sat())
+            .sum();
+        let target_sats = amount.to_sat();
 
-        // Assert: fidelity, outgoing swapcoins, hashlock and timelock coins are not included
-        assert!(
-            unspents.iter().all(|(_, spend_info)| !matches!(
-                spend_info,
-                UTXOSpendInfo::FidelityBondCoin { .. }
-                    | UTXOSpendInfo::OutgoingSwapCoin { .. }
-                    | UTXOSpendInfo::TimelockContract { .. }
-                    | UTXOSpendInfo::HashlockContract { .. }
-            )),
-            "Fidelity, Outgoing Swapcoins, Hashlock and Timelock coins are not included in coin selection"
+        log::info!(
+            "Separated coin selection: target={} sats, regular_total={} sats, swap_total={} sats",
+            target_sats,
+            regular_total,
+            swap_total
         );
 
-        let change_weight = Weight::from_vb_unwrap(CHANGE_OUTPUT_WEIGHT);
+        // Try regular UTXOs first, then swap UTXOs
+        for (utxo_type, unspents) in [
+            ("regular", &available_regular_utxos),
+            ("swap", &available_swap_utxos),
+        ] {
+            if unspents.is_empty() {
+                continue;
+            }
 
-        let cost_of_change = {
-            let creation_cost = calculate_fee(change_weight.to_vbytes_ceil(), feerate as f32)?;
-            let future_spending_cost = calculate_fee(P2WPKH_INPUT_WEIGHT / 4, LONG_TERM_FEERATE)?;
-            creation_cost + future_spending_cost
-        };
+            // Assert: fidelity, outgoing swapcoins, hashlock and timelock coins are not included
+            assert!(
+                unspents.iter().all(|(_, spend_info)| !matches!(
+                    spend_info,
+                    UTXOSpendInfo::FidelityBondCoin { .. }
+                        | UTXOSpendInfo::OutgoingSwapCoin { .. }
+                        | UTXOSpendInfo::TimelockContract { .. }
+                        | UTXOSpendInfo::HashlockContract { .. }
+                )),
+                "Fidelity, Outgoing Swapcoins, Hashlock and Timelock coins are not included in coin selection"
+            );
```

Comment:
We can perform this check before the loop, instead of inside it. It will reduce computation a bit. 

Should be done at the very start, where utxos are being fetched.